packages/types/src/primitive/Method.ts

```diff
@@ -65,7 +67,25 @@ export class MethodIndex extends U8aFixed {
 export default class Method extends Struct implements IMethod {
   protected _meta: FunctionMetadataV5;
 
-  constructor (value: any, meta?: FunctionMetadataV5) {
+  /**
+   * Method constructor
+   *
+   * @param value
+   * @param meta - Metadata to use, so that the `injectMethods` lookup is not
+   * necessary. This argument will soon be required so as to get rid of globals.
+   */
+  constructor (value: any, metaOrMethod?: FunctionMetadataV5 | MethodFunction) {
```

Comment:
In extension, you changed to https://github.com/polkadot-js/extension/blob/8d9463878602d59e6bca313ae8b49f01ee9365e0/packages/extension-ui/src/Popup/Signing/Details.tsx#L33

```javascript
   const method = new Method(data, Method.findMetaByValue(data, meta));
```

Instead of `FunctionMetadataV5 | MethodFunction`, we could do `FunctionMetadataV5 | Metadata`? So that you would just

```javascript
   const method = new Method(data, meta); // reads better
```

Edit: or for better future-proofness, `const method = new Method(data, {meta});`? Not sure...

---

There's another reason I prefer to pass the whole Metadata to Method/Event/Extrinsic. It's that the api/the end-user doesn't need to know what to extract from Metadata to pass to Method. Just pass the whole thing, and Method itself should extract the relevant information it needs.

Reply:
future-proofness is a good idea! :+1: passing the runtime metadata to the Method constructor feels to me way more practical but somehow less clean/functional (i.e. now `types` is somehow tied to the structure of the runtime metadata). but let's go with ease of use

Reply:
> (i.e. now types is somehow tied to the structure of the runtime metadata).

I see what you mean. But I would argue that even with a `MethodFunction` passed into the constructor, it's still tied to the structure of the runtime Metadata, because `MethodFunction.meta` is a `FunctionMetadataV5`, which can change with different metadata version.